pkg/kubelet/kubelet.go

```diff
@@ -350,6 +352,31 @@ func NewMainKubelet(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps *Kub
 	if err != nil {
 		return nil, err
 	}
+	if len(thresholds) > 0 {
```

Comment:
Container manager is meant to validate node configuration wrt. to compute
resources. Eviction manager is meant to deal with evictions only. It should
not ideally care about system configuration.

On Tue, Sep 6, 2016 at 11:15 AM, Michael Taufen <notifications@github.com>
wrote:

> In pkg/kubelet/kubelet.go
> <https://github.com/kubernetes/kubernetes/pull/31996#discussion_r77689370>
> :
>
> > @@ -350,6 +352,31 @@ func NewMainKubelet(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps *Kub
> >  	if err != nil {
> >  		return nil, err
> >  	}
> > +	if len(thresholds) > 0 {
>
> @vishh <https://github.com/vishh> Any reason container manager would be
> more appropriate than the eviction manager? Eviction manager seems better
> to me as it already knows whether eviction is enabled, and wrt #31676
> (comment)
> <https://github.com/kubernetes/kubernetes/issues/31676#issuecomment-244496413> eviction_enabled
> && swap_enabled is the necessary condition for an error.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31996/files/8c8e2fdb12920f5d56f1546ee676d87d008afd21#r77689370>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvIKPG5y-TJLnaahckBgRCXGOtA6x3Qks5qna3YgaJpZM4Jz-N7>
> .
>